services/sensiolabs/sensiolabs.service.js

```diff
@@ -1,110 +1,258 @@
 'use strict'
 
-const LegacyService = require('../legacy-service')
-const { makeBadgeData: getBadgeData } = require('../../lib/badge-data')
+const Joi = require('joi')
+const BaseXmlService = require('../base-xml')
 const serverSecrets = require('../../lib/server-secrets')
 
-// This legacy service should be rewritten to use e.g. BaseJsonService.
-//
-// Tips for rewriting:
-// https://github.com/badges/shields/blob/master/doc/rewriting-services.md
-//
-// Do not base new services on this code.
-module.exports = class Sensiolabs extends LegacyService {
+const schema = Joi.object({
+  project: Joi.object({
+    'last-analysis': Joi.object({
+      status: Joi.string()
+        .regex(/ordered|running|measured|analyzed|finished/)
+        .required(),
+      grade: Joi.string()
+        .regex(/platinum|gold|silver|bronze|none/)
+        .allow('', null),
+      violations: Joi.object({
+        violation: Joi.array()
+          .optional()
+          .items(
+            Joi.object({
+              severity: Joi.string()
+                .regex(/info|minor|major|critical/)
+                .required(),
+            })
+          ),
+      })
+        .optional()
```

Comment:
Still trying to validate what the xml response structure looks like during various conditions (analysis in progress, analysis completed but no violations found, etc.). This covers all the possibilities based on the xml response node structure, but I may be able to tighten it up more